#include "Exceptions.h"


//-------------------------------------------------------------------------------------------------
Exception::Exception(QString sMsg)
    :std::exception()
    ,m_sMsg(sMsg)
{}

//-------------------------------------------------------------------------------------------------
Exception::~Exception() throw()
{}

//-------------------------------------------------------------------------------------------------
const QString& Exception::getMessage() const throw()
{
    return m_sMsg;
}

//-------------------------------------------------------------------------------------------------
const char* Exception::what() const throw()
{
    return "";
}

//-------------------------------------------------------------------------------------------------
//
// FileIOException
//
//-------------------------------------------------------------------------------------------------

FileIOException::FileIOException(QString sMsg)
    :Exception(sMsg)
{}

//-------------------------------------------------------------------------------------------------
FileIOException::~FileIOException() throw()
{}

//-------------------------------------------------------------------------------------------------
//
// Internal Error
//
//-------------------------------------------------------------------------------------------------

InternalError::InternalError(QString sMsg)
    :Exception(sMsg)
{}

//-------------------------------------------------------------------------------------------------
InternalError::~InternalError() throw()
{}

//-------------------------------------------------------------------------------------------------
//
// Internal Error
//
//-------------------------------------------------------------------------------------------------

LuaException::LuaException(QString sMsg)
    :Exception(sMsg)
{}

//-------------------------------------------------------------------------------------------------
LuaException::~LuaException() throw()
{}
